packages/rest/src/rest.server.ts

```diff
@@ -683,11 +683,11 @@ export class RestServer extends Context implements Server, HttpServerLike {
    */
   redirect(
     fromPath: string,
-    toPathOrUrl: string,
+    toPathOrUrl: string | Function,
     statusCode?: number,
   ): Binding {
     return this.route(
-      new RedirectRoute(fromPath, this._basePath + toPathOrUrl, statusCode),
+      new RedirectRoute(fromPath, toPathOrUrl, statusCode),
```

Comment:
I think you should pass `this._basePath` to `RedirectRoute` constructor via a new argument.

Reply:
Even better: wait until https://github.com/strongloop/loopback-next/pull/2547 is landed and them modify RedirectRoute's `invokeHandler` function to obtain the base path from the request context via `requestContext.basePath`.